temporal-sdk/src/main/java/io/temporal/worker/WorkerFactory.java

```diff
@@ -393,6 +397,24 @@ public String toString() {
         "WorkerFactory{identity=%s, uniqueId=%s}", workflowClient.getOptions().getIdentity(), id);
   }
 
+  private static WorkflowThreadExecutor newWorkflowThreadExecutor(
+      ThreadPoolExecutor workflowThreadPool, Scope metricsScope) {
```

Comment:
Would it make sense for this to exist next to / inside the WorkflowThreadExecutor definition?

Reply:
I don't think so. It uses the constants and metrics-related logic that is not reusable and much more appropriate to be in this class.